satpy/readers/modis_l2.py

```diff
@@ -168,6 +168,34 @@ def get_dataset(self, dataset_id, dataset_info):
 
         return dataset
 
+    def available_datasets(self, configured_datasets):
+        """
+        Adds dataset information not specifically specified in reader yaml file
+        from arbitrary modis level 2 product files to available datasets.
+
+        Notes:
+             Currently only adds 2D datasets and does not decode bit encoded information.
+        """
+        # pass along existing datasets
+        for is_avail, ds_info in (configured_datasets or []):
+            yield is_avail, ds_info
+
+        res_dict = {(8120, 5416): 250, (4060, 2708): 500, (2030, 1354): 1000, (406, 270): 5000, (203, 135): 10000}
```

Comment:
There isn't an attribute in the file to determine this? I know some groups who modify the size of MODIS files (iirc) to different time ranges so these number of rows won't be consistent. Number of columns should be good enough, right?

Reply:
Unfortunately not at least I didn't find anything. I am not happy about this myself. If someone has a good idea for a smarter way to handle this I'd be very happy to change this.
Probably number of columns would be enough, will change it.